### ISSUE TITLE
clone help and output~ abstraction

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#1315]
[clone] is the only one that uses the output~ abstraction found in the audio examples. 

The way it is called, though, generates a very large GOP area.

see:

<img width="438" alt="Screen Shot 2021-05-19 at 02 30 08" src="https://user-images.githubusercontent.com/5360142/118761224-6c5c2180-b84a-11eb-963b-3b8020fa4131.png">

while the abstraction was meant to look like this 

<img width="379" alt="Screen Shot 2021-05-19 at 02 30 22" src="https://user-images.githubusercontent.com/5360142/118761260-7c740100-b84a-11eb-975d-ef5f339cfc4f.png">

One way would be to hide the object name in the settings, but that's bad I think.

Another would be to use [declare -path ../3.audio.examples/], which works but I think it's not an elegant solution.

The only other way would be to include the output~ abstraction in the same folder. This would create a third copy of this abstraction, which is now also part of the data structures examples and used only once in 07.sequencer.pd

Again, only clone uses this, so it'd be yet another copy just to be used once. Nonetheless I like this option and would like to use it. Moreover, I think I can find more excuses to use it in other examples of the help file.

**Comment by umlaeute** [comment]
> Another would be to use [declare -path ../3.audio.examples/], which works but I think it's not an elegant solution.

why?

**Comment by umlaeute** [comment]
> The only other way would be to include the output~ abstraction in the same folder. 

the only *other* other way, is to include the `[output~]` abstraction in the `extra/`-folder (but don't hold you breath, i'm sure i can come up with more other ways).

iirc, this has been discussed (or at least proposed) before, and people were not exactly happy with this due to nameclashes with your own `[output~]` abstraction (which might not be a GOP for *stereo* output at all). otoh `extra/` is searched last, so the spillover of Pd's `[output~]` abstraction into your own patches should be minimal. i guess the problem is more that *if* you have your own `[output~]` abstraction in your path, it would spill over to Pd's tutorials and help-patches (and eventually render them unusable). the only remedy to which would be to add a `[declare -stdpath .]` - which would finally be a valid use-case for the `-stdpath` flag of declare; but but we would need to add this to each and every tutorial/help-patch in Pd's standard distribution (which might actually not be such a bad idea)

**Comment by porres** [comment]
What I don't like about that declare use is that we're declaring a rather silly folder and I just prefer the idea of just including it in the folder - and we have a precedent of doing exactly that with the data structures' tutorial. 

> the only other other way, is to include the [output~] abstraction in the extra/-folder 

I thought about it, but didn't think the audio example's output~ deserved this promotion of becoming an 'extra' external (with a help file and all) cause it just seems sooooo simple... but... conceptually, that'd be the best strategy I think, other than having 3 copies of the same abstraction. And also, people could just use it when loading a patch that uses an output~ external they don't have.

> people were not exactly happy with this due to nameclashes with your own [output~] abstraction 

"my" output~ (from ELSE)? 

> (which might not be a GOP for stereo output at all)

"mine" is though and it's just the one we had in extended and I made some modifications. I actually did it for Purr Data and carry it in my library as an 'extra' as well. I don't actually use it and thought multiple times about just deleting it. I have the out~ object that I really use instead. But anyway, many patches from the extended era used this, like in help files of external libraries...

> otoh extra/ is searched last 

and quite unfortunately :/ but it's no use crying about it.

> the only remedy to which would be to add a [declare -stdpath .]

that's acceptable I guess

**Comment by porres** [comment]
This doesn't look so bad :) but I'd rather do this in a separate moment, for now I'll keep https://github.com/porres/pure-data/commit/4ea09245a94f65ea8e7721050771d0a843b048e8 and include output~ in other help files. At the end I can do this overall change. And we can also create a new issue just for this.  

<img width="1036" alt="Screen Shot 2021-05-19 at 12 15 04" src="https://user-images.githubusercontent.com/5360142/118838299-f1235b80-b89b-11eb-9127-af1572a48041.png">
